### ISSUE TITLE
preview app modules set-default launches browser for oauth when using a json key file

### ISSUE THREAD

**Issue by GoogleCodeExporter** [prototipando/google-cloud-sdk#133]
```
What steps will reproduce the problem?

google-cloud-sdk/bin/gcloud auth activate-service-account --key-file 
"${GAE_CLIENT_KEY_JSON_FILE}"
google-cloud-sdk/bin/gcloud \
    --project "${GAE_PROJECT_ID}" \
    preview app deploy \
    --version "${GAE_VERSION_LABEL}" \
    --quiet \
    app.yaml

google-cloud-sdk/bin/gcloud \
    --project "${GAE_PROJECT_ID}" \
    preview app modules set-default default \
    --quiet \
    --version "${GAE_VERSION_LABEL}"

What is the expected output? What do you see instead?

Version ${GAE_VERSION_LABEL} should be set as default, instead I see this:

Activated service account credentials for: 
[86709905025-k0cnttqgbs9eokapgoggm7mvk54jha5l@developer.gserviceaccount.com]
You are about to deploy the following modules:
 - kapx-site-demo/default/release-20150428  From: [/home/travis/build/KAPx/kapx-site/tmp_app.yaml]
Updating module [default] from file 
[/home/travis/build/KAPx/kapx-site/tmp_app.yaml]
06:05 PM Host: appengine.google.com
06:05 PM Application: kapx-site-demo (was: None); version: release-20150428 
(was: None)

... deployment happens here......

WARNING: This function, oauth2client.tools.run(), and the use of the gflags 
library are deprecated and will be removed in a future version of the library.
Your browser has been opened to visit:

    https://accounts.google.com/o/oauth2/auth?scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fappengine.admin&redirect_uri=http%3A%2F%2Flocalhost%3A8080%2F&response_type=code&client_id=32555940559.apps.googleusercontent.com&access_type=offline

If your browser is on a different machine then exit and re-run
this application with the command-line parameter 

  --noauth_local_webserver



No output has been received in the last 10 minutes, this potentially indicates 
a stalled build or something wrong with the build itself.

The build has been terminated

Please provide any additional information below.

I am running this in travis-ci in after_success so I cannot easily get the 
gcloud info. The first step I take on build is google-cloud-sdk/bin/gcloud 
components update preview app --quiet to be sure I have the latest sdk.


```

Original issue reported on code.google.com by `jojohns...@kaplan.edu` on 22 Apr 2015 at 6:50

**Comment by GoogleCodeExporter** [comment]
```
Here is gcloud info from the travis server


Google Cloud SDK [0.9.57]

Platform: [Linux, x86_64]
Python Version: [2.7.9 (default, Feb  4 2015, 20:51:02)  [GCC 4.6.3]]
Site Packages: [Enabled]

Installation Root: [/home/travis/build/KAPx/kapx-site/google-cloud-sdk]
Installed Components:
  core: [2015.04.21]
  app-engine-python: [1.9.18]
  core-nix: [2014.10.20]
  app-engine-go-linux-x86_64: [1.9.18]
  app-engine-java: [1.9.18]
  preview: [2015.04.21]
  app: [2015.04.21]
System PATH: 
[/home/travis/virtualenv/python2.7.9/bin:/home/travis/.gimme/versions/go1.4.1.li
nux.amd64/bin:/opt/python/2.7.9/bin:/opt/python/2.6.9/bin:/opt/python/3.4.2/bin:
/opt/python/3.3.5/bin:/opt/python/3.2.5/bin:/opt/python/pypy-2.5.0/bin:/opt/pyth
on/pypy3-2.4.0/bin:/usr/local/phantomjs/bin:/home/travis/.nvm/v0.10.36/bin:./nod
e_modules/.bin:/usr/local/maven-3.2.5/bin:/usr/local/clang-3.4/bin:/home/travis/
.gimme/versions/go1.4.1.linux.amd64/bin:/opt/python/2.7.9/bin:/opt/python/2.6.9/
bin:/opt/python/3.4.2/bin:/opt/python/3.3.5/bin:/opt/python/3.2.5/bin:/opt/pytho
n/pypy-2.5.0/bin:/opt/python/pypy3-2.4.0/bin:/usr/local/phantomjs/bin:./node_mod
ules/.bin:/usr/local/maven-3.2.5/bin:/usr/local/clang-3.4/bin:/home/travis/.gimm
e/versions/go1.4.1.linux.amd64/bin:/home/travis/.rvm/gems/ruby-1.9.3-p551/bin:/h
ome/travis/.rvm/gems/ruby-1.9.3-p551@global/bin:/home/travis/.rvm/rubies/ruby-1.
9.3-p551/bin:/opt/python/2.7.9/bin:/opt/python/2.6.9/bin:/opt/python/3.4.2/bin:/
opt/python/3.3.5/bin:/opt/python/3.2.5/bin:/opt/python/pypy-2.5.0/bin:/opt/pytho
n/pypy3-2.4.0/bin:/usr/local/phantomjs/bin:./node_modules/.bin:/usr/local/maven-
3.2.5/bin:/usr/local/clang-3.4/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr
/bin:/sbin:/bin:/home/travis/.rvm/bin:/home/travis/.rvm/bin:/home/travis/.rvm/bi
n]
Cloud SDK on PATH: [False]

Installation Properties: 
[/home/travis/build/KAPx/kapx-site/google-cloud-sdk/properties]
User Config Directory: [/home/travis/.config/gcloud]
User Properties: [/home/travis/.config/gcloud/properties]
Current Workspace: [None]
Workspace Config Directory: [None]
Workspace Properties: [None]

Account: 
[86709905025-k0cnttqgbs9eokapgoggm7mvk54jha5l@developer.gserviceaccount.com]
Project: [None]

Current Properties:
  [core]
    account: [86709905025-k0cnttqgbs9eokapgoggm7mvk54jha5l@developer.gserviceaccount.com]
    disable_usage_reporting: [True]

Logs Directory: [/home/travis/.config/gcloud/logs]
Last Log File: [/home/travis/.config/gcloud/logs/2015.04.23/19.15.56.912374.log]
```

Original comment by `jojohns...@kaplan.edu` on 23 Apr 2015 at 7:21

**Comment by GoogleCodeExporter** [comment]
```
I was able to recreate it with a new client secret json file on a box that 
never had gcloud installed before. It seems to be an issue with service 
accounts when you don't have a personal account set up. I cannot recreate it 
locally when I have my personal account authed as well.

Steps:
wget https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.zip;
unzip --qq google-cloud-sdk.zip
CLOUDSDK_CORE_DISABLE_PROMPTS=1 google-cloud-sdk/install.sh
google-cloud-sdk/bin/gcloud components update app pkg-core pkg-python preview 
gcloud gae-python --quiet
source .bashrc
gcloud auth activate-service-account --key-file kapx-site-demo.json
google-cloud-sdk/bin/gcloud components update app pkg-core pkg-python preview 
gcloud gae-python --quiet
gcloud --project kapx-site-demo preview app modules set-default default --quiet 
--version release-20150428

(opens web browser if present)

shell says:

WARNING: This function, oauth2client.tools.run(), and the use of the gflags 
library are deprecated and will be removed in a future version of the library.
Your browser has been opened to visit:

    https://accounts.google.com/o/oauth2/auth?scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fappengine.admin&redirect_uri=http%3A%2F%2Flocalhost%3A8090%2F&response_type=code&client_id=32555940559.apps.googleusercontent.com&access_type=offline

If your browser is on a different machine then exit and re-run
this application with the command-line parameter

  --noauth_local_webserver
```

Original comment by `jojohns...@kaplan.edu` on 27 Apr 2015 at 5:35

**Comment by GoogleCodeExporter** [comment]
```
Same problem here. 
Running "preview app deploy" tries to open browser -> NOK
Running "config list" shows the service account activated -> OK
Running "compute images list" list the images -> OK

The following jenkins log, shows the output of the above commands:
[easy-easy-apps-server_staging] $ gcloud auth activate-service-account 
994979732023-h4alem9cpokm2sgvdvfn0qo5ggu1gubg@developer.gserviceaccount.com 
--key-file 
/home/jenkins/workspace/easy-easy-apps-server_staging/gcloud4727254661203477121.
tmp/key8801487748596692342.json
Activated service account credentials for: 
[994979732023-h4alem9cpokm2sgvdvfn0qo5ggu1gubg@developer.gserviceaccount.com]
[easy-easy-apps-server_staging] $ /bin/sh -xe /tmp/hudson8106852726170846078.sh
+ cp docker/staging/Dockerfile .
+ cp docker/staging/app.yaml .
[easy-easy-apps-server_staging] $ gcloud config set project 
server-easyeasyapps-staging
[easy-easy-apps-server_staging] $ gcloud config list
[core]
account = 
994979732023-h4alem9cpokm2sgvdvfn0qo5ggu1gubg@developer.gserviceaccount.com
disable_usage_reporting = False
project = server-easyeasyapps-staging
[easy-easy-apps-server_staging] $ gcloud compute images list
NAME                                PROJECT           ALIAS              
DEPRECATED STATUS
centos-6-v20150325                  centos-cloud      centos-6                  
    READY
centos-7-v20150325                  centos-cloud      centos-7                  
    READY
coreos-alpha-660-0-0-v20150423      coreos-cloud                                
    READY
coreos-beta-647-0-0-v20150415       coreos-cloud                                
    READY
coreos-stable-633-1-0-v20150414     coreos-cloud      coreos                    
    READY
backports-debian-7-wheezy-v20150325 debian-cloud      debian-7-backports        
    READY
debian-7-wheezy-v20150325           debian-cloud      debian-7                  
    READY
container-vm-v20150129              google-containers container-vm              
    READY
container-vm-v20150305              google-containers container-vm              
    READY
container-vm-v20150317              google-containers container-vm              
    READY
opensuse-13-1-v20141102             opensuse-cloud    opensuse-13               
    READY
opensuse-13-2-v20150315             opensuse-cloud    opensuse-13               
    READY
rhel-6-v20150325                    rhel-cloud        rhel-6                    
    READY
rhel-7-v20150325                    rhel-cloud        rhel-7                    
    READY
sles-11-sp3-v20150310               suse-cloud        sles-11                   
    READY
sles-12-v20150310                   suse-cloud        sles-12                   
    READY
ubuntu-1204-precise-v20150316       ubuntu-os-cloud   ubuntu-12-04              
    READY
ubuntu-1404-trusty-v20150316        ubuntu-os-cloud   ubuntu-14-04              
    READY
ubuntu-1410-utopic-v20150318c       ubuntu-os-cloud   ubuntu-14-10              
    READY
ubuntu-1504-vivid-v20150422         ubuntu-os-cloud                             
    READY
windows-server-2008-r2-dc-v20150331 windows-cloud     windows-2008-r2           
    READY
windows-server-2012-r2-dc-v20150331 windows-cloud     windows-2012-r2           
    READY
[easy-easy-apps-server_staging] $ gcloud compute instances list
NAME               ZONE          MACHINE_TYPE  INTERNAL_IP   EXTERNAL_IP   
STATUS
gae-default-1-k8km us-central1-f n1-standard-1 10.240.212.26 104.197.28.31 
RUNNING
[easy-easy-apps-server_staging] $ gcloud --quiet --project 
server-easyeasyapps-staging preview app deploy app.yaml --version 1
WARNING: The [version] field is specified in file 
[/home/jenkins/workspace/easy-easy-apps-server_staging/app.yaml].  This field 
is not used by gcloud and should be removed.
You are about to deploy the following modules:
 - server-easyeasyapps-staging/default/1  From: [/home/jenkins/workspace/easy-easy-apps-server_staging/app.yaml]
Go to the following link in your browser:

    https://accounts.google.com/o/oauth2/auth?scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fappengine.admin&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&response_type=code&client_id=32555940559.apps.googleusercontent.com&access_type=offline

Enter verification code: Traceback (most recent call last):
  File "/usr/local/share/google-cloud-sdk/./lib/googlecloudsdk/gcloud/gcloud.py", line 211, in <module>
    main()
  File "/usr/local/share/google-cloud-sdk/./lib/googlecloudsdk/gcloud/gcloud.py", line 207, in main
    _cli.Execute()
  File "/usr/local/share/google-cloud-sdk/./lib/googlecloudsdk/calliope/cli.py", line 617, in Execute
    result = args.cmd_func(cli=self, args=args)
  File "/usr/local/share/google-cloud-sdk/./lib/googlecloudsdk/calliope/backend.py", line 1110, in Run
    result = command_instance.Run(args)
  File "/usr/local/share/google-cloud-sdk/./lib/googlecloudsdk/calliope/exceptions.py", line 86, in TryFunc
    return func(*args, **kwargs)
  File "/usr/local/share/google-cloud-sdk/lib/googlecloudsdk/appengine/app_commands/deploy.py", line 124, in Run
    client.PrepareVmRuntime()
  File "/usr/local/share/google-cloud-sdk/./lib/googlecloudsdk/appengine/lib/appengine_client.py", line 251, in PrepareVmRuntime
    rpcserver.Send('/api/vms/prepare', app_id=self.project)
  File "/usr/local/share/google-cloud-sdk/./lib/googlecloudsdk/appengine/lib/appengine_client.py", line 525, in Send
    response = self._server.Send(*args, **kwargs)
  File "/usr/local/share/google-cloud-sdk/./lib/googlecloudsdk/appengine/lib/external/tools/appengine_rpc_httplib2.py", line 214, in Send
    self._Authenticate(self.http, auth_errors[0] > 0)
  File "/usr/local/share/google-cloud-sdk/./lib/googlecloudsdk/appengine/lib/external/tools/appengine_rpc_httplib2.py", line 423, in _Authenticate
    self.credentials = tools.run_flow(flow, self.storage, self.flags)
  File "/usr/local/share/google-cloud-sdk/./lib/oauth2client/util.py", line 129, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/share/google-cloud-sdk/./lib/oauth2client/tools.py", line 208, in run_flow
    code = raw_input('Enter verification code: ').strip()
EOFError: EOF when reading a line
Build step 'Execute gcloud CLI' marked build as failure
[easy-easy-apps-server_staging] $ gcloud auth revoke 
994979732023-h4alem9cpokm2sgvdvfn0qo5ggu1gubg@developer.gserviceaccount.com
No credentialed accounts.

To login, run:
  $ gcloud auth login ``ACCOUNT''

Revoked credentials for 
994979732023-h4alem9cpokm2sgvdvfn0qo5ggu1gubg@developer.gserviceaccount.com.

```

Original comment by `jrl...@gmail.com` on 4 May 2015 at 9:37

**Comment by GoogleCodeExporter** [comment]
```
For what it's worth, the service account has the following scopes shown in 
.config/gcloud/credentials

"https://www.googleapis.com/auth/userinfo.email 
https://www.googleapis.com/auth/cloud-platform 
https://www.googleapis.com/auth/appengine.admin 
https://www.googleapis.com/auth/compute"
```

Original comment by `jojohns...@kaplan.edu` on 13 May 2015 at 6:10

**Comment by GoogleCodeExporter** [comment]
```
Here is an example you can use to see what I am doing:

This is the log using this shell script to deploy to gcloud on travis-ci 
https://github.com/Trii/appengine-buildout-template/blob/ef6b433e70051155295bfeb
f8f266b3d4a7b7b26/travis-ci-deploy.sh

Here is the log showing it prompting for oauth
https://travis-ci.org/Trii/appengine-buildout-template/builds/62617755


Removing the --set-default flag from gcloud preview app deploy 
https://github.com/Trii/appengine-buildout-template/blob/248650fe674dd81a6934615
f4ca77236d79bfef2/travis-ci-deploy.sh

ends in success:
https://travis-ci.org/Trii/appengine-buildout-template/builds/62618698

The same issue happens if I try to use preview app modules set-default after 
deploying, it prompts for oauth
```

Original comment by `jojohns...@kaplan.edu` on 14 May 2015 at 9:55

**Comment by GoogleCodeExporter** [comment]
```
We've fixed this in version 0.9.63, which is slated for release next week.

Thank you for the detailed feedback; it was very helpful in diagnosing the 
problem.
```

Original comment by `z...@google.com` on 28 May 2015 at 7:01
- Changed state: **Fixed**
